ethcore/src/miner/miner.rs

```diff
@@ -908,6 +908,14 @@ impl miner::MinerService for Miner {
 		}
 	}
 
+	fn clear_author(&self) {
+		self.params.write().author = Default::default();
+		if self.engine.sealing_state() == SealingState::Ready {
+			self.sealing.lock().enabled = false;
+			self.engine.clear_signer();
```

Comment:
I don't understand why you are only `clearing` engine signer when `SealingState::Ready`? Can you explain thanks?

Reply:
I agree with you it's convoluted and I'm open to suggestions.

1. Should we disable sealing unconditionally?
2. Or should we not disable sealing at all in `clear_author`, possibly requiring a separate call to `clear_signer`?

Reply:
I will change this to clearing the signer when the sealing state is other than `SealingState::External`. This will make `clear_author` consistent with `set_author`.